theano/compile/function.py

```diff
@@ -267,4 +295,16 @@ def opt_log1p(node):
     # We need to add the flag check_aliased inputs if we have any mutable or
     # borrowed used defined inputs
     fn._check_for_aliased_inputs = check_for_aliased_inputs
+
+    if outputs_dict_format:
+
+        fn_dict_output = (lambda *args, **kwargs: output_dictionary_wrapper(args, kwargs, fn = fn, keys = output_keys))
```

Comment:
I think this work, but I think it break the pickling of Theano function that use this feature. Also, it break the assumption that what we return is an instance of Function and so it do not have the attribute maker for example.

I think this code should be done in the method Function.__call__ directly. That should fix the pickling problem and keep the same object with the same attribute.

Can you add a test that make sure we do not break this mean feature?

thanks